src/issues/currentIssue.ts

```diff
@@ -183,75 +198,124 @@ export class CurrentIssue extends Disposable {
-		const branchNameConfig = await variableSubstitution(
-			await this.getBranchTitle(),
-			this.issue,
-			undefined,
-			await this.getUser(),
-		);
-		const branchNameMatch = this._branchName?.match(new RegExp('^(' + branchNameConfig + ')(_)?(\\d*)'));
-		if ((createBranchConfig === 'on')) {
-			const branch = await this.getBranch(this._branchName!);
-			if (!branch) {
-				if (!branchNameMatch) {
-					this._branchName = branchNameConfig;
-				}
-			} else if (!silent) {
-				this._branchName = await this.offerNewBranch(branch, branchNameConfig, branchNameMatch);
+		const issueNumberStr = this.issueModel.number.toString();
+		const issueTitle = this.issueModel.title;
+		const suggestedBranchName = (await this.getBranchTitle()).toLocaleLowerCase();
+
+		this._branchName = undefined;
+
+		const branches = await this.manager.repository.getBranches({ remote: true });
+		const branchesWithIssueMatch: vscode.QuickPickItem[] = [];
+		const otherBranches: vscode.QuickPickItem[] = [];
+
+		branches.forEach((branch) => {
+			const isRemote = branch.name?.startsWith('origin/');
+			const displayName = branch.name ?? '';
+			const branchItem = {
+				label: `${isRemote ? '$(cloud)' : '$(git-branch)'} ${displayName}`,
+				description: `${isRemote ? 'Remote' : 'Local'} branch`,
+			};
+
+			if (
+				branch.name?.toLowerCase().includes(issueNumberStr.toLowerCase()) ||
+				branch.name?.toLowerCase().includes(issueTitle.replace(/[^a-zA-Z0-9-]/g, '-').toLowerCase()) ||
+				branch.name?.toLowerCase().includes(issueTitle.replace(/[^a-zA-Z0-9-]/g, '_').toLowerCase())
+			) {
+				branchesWithIssueMatch.push(branchItem);
+			} else {
+				otherBranches.push(branchItem);
 			}
+		});
+
+		// Create QuickPick items
+		const branchItems: vscode.QuickPickItem[] = [
+			{ label: 'Suggested branch', kind: vscode.QuickPickItemKind.Separator },
+			{
+				label: `$(lightbulb) ${suggestedBranchName}`,
+				description: '',
+				detail: 'Recommended branch name based on settings',
+				picked: true,
+			},
+		];
+
+		if (branchesWithIssueMatch.length > 0) {
+			branchItems.push({ label: 'Branches matching', kind: vscode.QuickPickItemKind.Separator });
+			branchItems.push(...branchesWithIssueMatch);
 		}
-		if (!this._branchName) {
-			this._branchName = await vscode.window.showInputBox({
-				value: branchNameConfig,
-				prompt: vscode.l10n.t('Enter the label for the new branch.'),
-			});
+
+		if (otherBranches.length > 0) {
+			branchItems.push({ label: 'Other branches', kind: vscode.QuickPickItemKind.Separator });
+			branchItems.push(...otherBranches);
 		}
-		if (!this._branchName) {
-			// user has cancelled
+
+		branchItems.push(
+			{ label: 'Custom branch name:', kind: vscode.QuickPickItemKind.Separator },
+			{ label: '$(pencil) Enter a custom branch name...', description: 'Choose this to type your own branch name' }
+		);
+
+		// Show QuickPick for branch selection
+		const quickPick = vscode.window.createQuickPick<vscode.QuickPickItem>();
```

Comment:
Can you just use `showQuickPick` here instead of the more complicated `createQuickPick`?